### ISSUE TITLE
Objects disappearing

### ISSUE THREAD

**Issue by BartNWD** [samp-incognito/samp-streamer-plugin#309]
Hello,

I've been using Incognito's plugin since 2012, and I've always had this problem in my script. I have many dynamic objects (and static objects too, CreateObject I use for objects that can't be streamed, such as floors to avoid vehicles falling through the dynamic objects). I have systems where players can place furniture objects in their houses, police men can place roadblocks, and admins can place objects around the map. They all work with dynamic objects, but when someone places a roadblock, furniture, or object, the object just dissapears after a few seconds.

It's not destroyed by the script, it just dissapears. Oddly enough, this only happens with objects created on the run, and not OnGameModeInit (initial maps), such as the ones I mentioned, roadblocks and furniture and dynamic objects placed by admins. This isn't a limit problem, since this also happens when placing an object in the desert with nothing scripted there.

I have a Ubuntu server in the latest streamer version. This problem is mortifying me and I am looking for a solution. 

Thanks, beforehand.

**Comment by IstuntmanI** [comment]
Make sure that:
1. You are using one of the latest Streamer Plugin versions.
2. You recompiled all of your scripts (gamemode + filterscripts) with the correct include - in some past SA-MP version the `drawdistance` was messed up by some change and everything had to be recompiled, but this is probably not the case here.
3. You aren't actually creating too many global objects (`CreateObject`) - this may explain why they go away in desert.
4. There's actually no difference between dynamic objects created in `OnGameModeInit` or objects created anywhere else. When `CreateDynamicX` is called, the item is simply added in the memory and then, an internal timer checks if a player is close to the position of any object position added in memory. So, because of this, it may be possible that in `OnGameModeInit` you create them with some fine settings (such as correct `streamdistance` and `drawdistance`), while in commands, dialogs and other things you are creating them with weird and incorrect properties (`streamdistance` too small, maybe, or any other similar thing)

**Comment by BartNWD** [comment]
In total, there are 7 CreateObject, the bug is actually the plugin, and I fight with it a long time. All scripts are compiled in the correct version.

I changed to version 2.8.2 recently, and it looks a little better. It takes time to happen, but it happens the same way. But this version does not have DynamicActors and I had to pass all my systems to the CreateActor.

Another thing I noticed is that when the object disappears, its rotation is modificated. For example, a player's balloon, when created, stay in normal position 10 seconds, is placed in the opposite rotation, lying on the ground and then disappears. This is very fast, matter of milliseconds between changing the rotation and disappearing.

Here are some videos:

https://www.youtube.com/watch?v=xCUTFcLqGsk
https://www.youtube.com/watch?v=Lbpzl2oV9TQ

**Comment by samp-incognito** [comment]
Can you reproduce that in an empty script?

**Comment by BartNWD** [comment]
I've tried filterscripts, clean gamemodes, and the same thing happens. With few players this does not happen.

**Comment by aktah** [comment]
This problem also happens to me. Create furniture on different virtual world, The object dissapears after a few seconds. The problem is not going to happen if there are only a few players.

**Comment by MacLove13** [comment]
Have an solution?

**Comment by Darkeen** [comment]
Same on v2.9.2 and v2.9.3.

**Comment by Gpro25** [comment]
Same, still fighting with this for months..

**Comment by Hoxxy** [comment]
Also having this issue.

I create around 18000 objects at server startup, and they never disappear. However, objects that are added after startup (such as house furnishing) do disappear after some time, one by one, in order they were created.
Reloading those objects from the database and creating them again doesn't fix the issue - they still disappear sooner or later. 

@samp-incognito @IstuntmanI any way to fix this?

**Comment by samp-incognito** [comment]
I'm almost positive it's because you're getting object IDs mixed up somewhere. This is the most common scripting problem I've seen, especially when an object is destroyed and the old ID continues to reference newly-created objects. The fact that the objects don't "disappear" if they're created at startup is telling. I'm assuming they aren't assigned to any variables and thus never get modified in any way. Check your script carefully to see how you're handling IDs.

Otherwise, if it truly is a plugin problem, then the only way it can be fixed is if you've somehow found a way to reproduce it.

**Comment by markski1** [comment]
Starting to get this in v2.9.2 after some updates to the server.

The following changes were made:
- discord connector upgraded to latest version (shouldn't affect streamer)
- MathPlugin removed (shouldn't affect streamer)
- Updated server's kernel from 3.9 to 4.19
- Updated server from Debian Jessie to Buster.

No changes to the script were made other than Discord connector stuff. Object-related code was not touched in months. However, after doing the updates above, none of which is related to Streamer, objects that were spawned after OnGameModeInit began to randomly dissapear (or, more specifically, they're replacing each other). I've witnessed very weird bugs such as objects that were created elsewhere in the map dissapearing and reappearing where a new object should've been created. 

For example, I saw a player have a drug plant from somewhere else in the map appear on top of their vehicle, instead of what should've been a newly created arrow.

Considering that all of this worked correctly for months before the updates in question I think something is messing with the plugin directly. I don't know how to reproduce it, it just happens. @samp-incognito

**Comment by BartNWD** [comment]
Hello,

After 2 years of creating this issue, I can confirm that this problem continues to happen with objects created after the server started. With objects attached to vehicles and players mainly, but also with objects like life machines, balloons, and more.

Few months ago, we solved this problem with another problem, using this:

`CreateDynamicObject(19300,0.00000000,0.00000000,0.00000000.0.00000000.0.00000000.0.00000000); // Invalid Object ID (Streamer bug)`

However, this is not solving the problem anymore. @samp-incognito

**Comment by Ranneciva** [comment]
Hey, I got this problem too. Have you got any ideas to fix? @BartNWD

**Comment by BartNWD** [comment]
No, until today the problem has not been solved. On the contrary, it gets more and more strange and random. I tested old versions of the plugin, since 2.7.2, and the problem persists.

Some more strange happenings that I captured:

_Balloons creating on PlayerAttachedObject slot:_ 
https://www.youtube.com/watch?v=9011-OFezZ0

_After disappearing, go to point 0: (using / goto object command)_
https://www.youtube.com/watch?v=qCUOWeW0mz4

_The fire when firing nitro in vehicles never stabilizes:_
https://www.youtube.com/watch?v=euT_EJAqe0U

NOTE: The smaller the number of players, the fewer occurrences. With 1 player everything just happens perfectly. In version 2.7.2 it is also possible to notice a reduction in occurrences.

Note²: I use streamer in gamemode and 2 filterscripts.

**Comment by Ranneciva** [comment]
Me too. I got this problem only above 100 players in my server. And this happened. @BartNWD.
So, how you solve this for this time?

**Comment by BartNWD** [comment]
> Me too. I got this problem only above 100 players in my server. And this happened. @BartNWD.
> So, how you solve this for this time?

I didn't solve :c

**Comment by Hauswart** [comment]
Here's a tip!
For localobjects use player names.

**Comment by Hauswart** [comment]
> Note²: I use streamer in gamemode and 2 filterscripts.

ThIs is the problem.

**Comment by Kutuy** [comment]
> > Me too. I got this problem only above 100 players in my server. And this happened. @BartNWD.
> > So, how you solve this for this time?
> 
> I didn't solve :c

Have you solved this issue?

**Comment by raedman90** [comment]
I'm experiencing the same problem, apparently with no effective resolution yet.

**Comment by meimoi** [comment]
> Hello,
> 
> I've been using Incognito's plugin since 2012, and I've always had this problem in my script. I have many dynamic objects (and static objects too, CreateObject I use for objects that can't be streamed, such as floors to avoid vehicles falling through the dynamic objects). I have systems where players can place furniture objects in their houses, police men can place roadblocks, and admins can place objects around the map. They all work with dynamic objects, but when someone places a roadblock, furniture, or object, the object just dissapears after a few seconds.
> 
> It's not destroyed by the script, it just dissapears. Oddly enough, this only happens with objects created on the run, and not OnGameModeInit (initial maps), such as the ones I mentioned, roadblocks and furniture and dynamic objects placed by admins. This isn't a limit problem, since this also happens when placing an object in the desert with nothing scripted there.
> 
> I have a Ubuntu server in the latest streamer version. This problem is mortifying me and I am looking for a solution.
> 
> Thanks, beforehand.

Yo' my friend
Have you fixed this issue?
